src/main/java/org/ibp/api/brapi/v2/observationunits/ObservationUnitResourceBrapi.java

```diff
@@ -92,45 +100,81 @@
 				HttpStatus.NOT_FOUND);
 		}
 
 		final ModelMapper mapper = ObservationUnitMapper.getInstance();
-		final PhenotypeSearchRequestDTO phenotypeSearchRequestDTO = mapper.map(observationUnitsSearchRequestDto, PhenotypeSearchRequestDTO.class);
+		final ObservationUnitSearchRequestDTO phenotypeSearchRequestDTO =
+			mapper.map(observationUnitsSearchRequestDto, ObservationUnitSearchRequestDTO.class);
 
-		final PagedResult<PhenotypeSearchDTO> resultPage = this.getObservationUnitDtoPagedResult(phenotypeSearchRequestDTO, currentPage, pageSize);
+		final PagedResult<ObservationUnitDto> resultPage =
+			this.getObservationUnitDtoPagedResult(phenotypeSearchRequestDTO, currentPage, pageSize);
 
-		final Result<PhenotypeSearchDTO> results = new Result<PhenotypeSearchDTO>().withData(resultPage.getPageResults());
+		final Result<ObservationUnitDto> results = new Result<ObservationUnitDto>().withData(resultPage.getPageResults());
 		final Pagination pagination = new Pagination().withPageNumber(resultPage.getPageNumber()).withPageSize(resultPage.getPageSize())
 			.withTotalCount(resultPage.getTotalResults()).withTotalPages(resultPage.getTotalPages());
 
 		final Metadata metadata = new Metadata().withPagination(pagination);
 
-		final EntityListResponse<PhenotypeSearchDTO> entityListResponse = new EntityListResponse<>(metadata, results);
+		final EntityListResponse<ObservationUnitDto> entityListResponse = new EntityListResponse<>(metadata, results);
 
 		return new ResponseEntity<>(entityListResponse, HttpStatus.OK);
 
 	}
 
-	private PagedResult<PhenotypeSearchDTO> getObservationUnitDtoPagedResult(
-		final PhenotypeSearchRequestDTO phenotypeSearchDTO,
+	@ApiOperation(value = "Add new Observation Units", notes = "Add new Observation Units")
+	@RequestMapping(value = "/{crop}/brapi/v2/observationunits", method = RequestMethod.POST)
+	@ResponseBody
+	@JsonView(BrapiView.BrapiV2.class)
+	public ResponseEntity<EntityListResponse<ObservationUnitDto>> createObservationUnits(@PathVariable final String crop,
+		@RequestBody final List<ObservationUnitImportRequestDto> observationUnitImportRequestDtos) {
+		BaseValidator.checkNotNull(observationUnitImportRequestDtos, "observation.unit.import.request.null");
+
+		final ObservationUnitImportResponse observationUnitImportResponse =
+			this.observationUnitService.createObservationUnits(crop, observationUnitImportRequestDtos);
+		final Result<ObservationUnitDto> results =
+			new Result<ObservationUnitDto>().withData(observationUnitImportResponse.getObservationUnits());
+
+		final List<Map<String, String>> status = new ArrayList<>();
+		final Map<String, String> messageInfo = new HashMap<>();
+		messageInfo.put("message", observationUnitImportResponse.getStatus());
+		messageInfo.put("messageType", "INFO");
+		status.add(messageInfo);
+		if (!CollectionUtils.isEmpty(observationUnitImportResponse.getErrors())) {
+			int index = 1;
+			for (final ObjectError error : observationUnitImportResponse.getErrors()) {
```

Comment:
Use ResponseMessageGenerator.getMessagesList() to generate message status list. Similar to what's implemented in POST /{crop}/brapi/v2/studies